libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -383,6 +383,7 @@ void DeclarationTypeChecker::endVisit(VariableDeclaration const& _variable)
 					string(_variable.isReturnParameter() ? "return " : "") +
 					"parameter in" +
 					string(_variable.isExternalCallableParameter() ? " external" : "") +
+					string(_variable.isPublicCallableParameter() ? " public" : "") +
```

Comment:
This is correct but I think we should also be doing this for return parameters. I see a lot of messages in tests that still say "in function" and those are either for storage return parameters or for cases where a parameter has no location at all.

Actually, I see no reason not to make it simple and just always include the visibility in the message. You can get visibility like this (not tested but should work):
```c++
string visibility;
if (auto const* callable = dynamic_cast<CallableDeclaration const*>(_variable.scope()))
    visibility = Declaration::visibilityToString(callable->visibility());
```